core/modules/server/server-sent-events.js

```diff
@@ -0,0 +1,65 @@
+/*\
+title: $:/core/modules/server/server-sent-events.js
+type: application/javascript
+module-type: library
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+parameters:
+		prefix - usually the plugin path, such as `plugins/tiddlywiki/tiddlyweb`. The
+			route will match `/events/${prefix}` exactly.
+
+		handler - a function that will be called each time a request comes in with the
+			request and state from the route and an emit function to call.
+*/
+
+var ServerSentEvents = function ServerSentEvents(prefix, handler) {
+  this.handler = handler;
+  this.prefix = prefix;
+};
+
+ServerSentEvents.prototype.getExports = function() {
+  return {
+    bodyFormat: "stream",
+    method: "GET",
+    path: new RegExp("^/events/" + this.prefix + "$"),
+    handler: this.handleEventRequest.bind(this)
+  };
+};
+
+ServerSentEvents.prototype.handleEventRequest = function(request,response,state) {
+  if(request.headers.accept && request.headers.accept.startsWith("text/event-stream")) {
```

Comment:
`startsWith` is an ECMAScript 2015 thing, we try to stick to 5.1.

Reply:
Done.